client/sniffer/src/main/java/org/elasticsearch/client/sniff/Sniffer.java

```diff
@@ -116,15 +118,19 @@ void sniffOnFailure(HttpHost failedHost) {
         void sniff(HttpHost excludeHost, long nextSniffDelayMillis) {
             if (running.compareAndSet(false, true)) {
                 try {
-                    List<HttpHost> sniffedHosts = hostsSniffer.sniffHosts();
+                    final List<Node> sniffedHosts = hostsSniffer.sniffHosts();
                     logger.debug("sniffed hosts: " + sniffedHosts);
                     if (excludeHost != null) {
-                        sniffedHosts.remove(excludeHost);
+                        for (Iterator<Node> itr = sniffedHosts.iterator(); itr.hasNext();) {
+                            if (itr.next().getHost().equals(excludeHost)) {
+                                itr.remove();
+                            }
+                        }
                     }
                     if (sniffedHosts.isEmpty()) {
                         logger.warn("no hosts to set, hosts will be updated at the next sniffing round");
                     } else {
-                        this.restClient.setHosts(sniffedHosts.toArray(new HttpHost[sniffedHosts.size()]));
+                        this.restClient.setNodes(sniffedHosts.toArray(new Node[0]));
```

Comment:
can you use the known size rather than `0`?

Reply:
I can but I'm not [sure](https://shipilev.net/blog/2016/arrays-wisdom-ancients/) it is better. I tend to just use 0 because it is simpler but I didn't mean to change what was there.

Reply:
ok :)